flang-rt/lib/FortranFloat128Math/CMakeLists.txt

```diff
@@ -15,6 +15,7 @@
 # used for building this FortranFloat128Math library.
 
 include(CheckLibraryExists)
+include(CheckIncludeFile)
```

Comment:
5 out my test configurations are also using ` -DFLANG_RUNTIME_F128_MATH_LIB=libquadmath`. I did not see such error yet. I am curious about your findings.

Reply:
So my issue is indeed related to the bootsrap build of the runtime: the gcc compiler I use to build flang/llvm (the in my cmake command) knows how to find quadmath.h without hints because this header sits in this gcc compiler toochain include directory. When reaching the flang-rt build step, the build switch to using the just built clang that does not know how to find it.

That explains why I never had the issue in out-of-tree builds where I controlled the runtime compiler and flags.

Three questions:
- Is it possible to tell llvm to build flang-rt with the same compiler and options than the one from flang (instead of bootstraping)?
- In bootstrap builds, how can one specify compiler options for the runtime build steps, but not the compiler build? That way I could point the gcc toolchain I want to use to clang for the second step.
- Is there a drawback in always building the runtime out-of-tree instead of using bootstrap builds, apart from the extra cmake command (i.e., are there some features/driver comfiguration step that only happens in bootstrapped builds)?

Reply:
1. In bootstrap builds, using just-built Clang/Flang as compiler (`CMAKE_*_COMPILER`) is hardcoded (that's where the name bootstrap comes from). You want a runtimes-only build here. Runtimes like offload could not even work with gcc since those generate LLVM-bc files for link-time compilation. To unify offloading accross runtimes, I think @jhuber6 would like to do something similar for Flang-RT.

2. Using `-DLLVM_ENABLE_RUNTIMES_PASSTHROUGH=ON` you can specify additional CMake variables that should be forwared to the runtimes build configuration. Specifying `CMAKE_CXX_FLAGS` here should be possible. Also, there is a [mechanism](https://github.com/llvm/llvm-project/blob/main/llvm/runtimes/CMakeLists.txt#L369-L393) to forward variables prefixed with `RUNTIMES_` or listed in `FLANG_RT_extra_args`, but I have never used it.

3. Not that I know of, but in my experience getting the configurations to match can easily go wrong.

I assume that you are going to try to point Clang to the location of gcc's `quadmath.h` and `libquadmath.a`? Clang autodetect a gcc installation and usually is able to reuse its files. Don't now why this doesn't happen for quadmath.

Note that relying on compiler and runtime having the same value of `FLANG_RUNTIME_F128_MATH_LIB` is something we should get away from, especially for cross-compilation. There is no guarantee that the target platform has libquadmath support only because the host platform has.